### **5 dakika içinde veriler nasıl filitreleneri öğrenin!**

![](https://miro.medium.com/max/1400/0*LLbrg6Gicb6JxQve)

 [Unsplash](https://unsplash.com/?utm_source=medium&utm_medium=referral)'de [Adam Jaime](https://unsplash.com/@arobj?utm_source=medium&utm_medium=referral) tarafından bir fotoğraf

## İçindekiler
- **Belirli sütunları seçme**
- **Belirli sütunları ve satırları seçme**
- **Bir sütundaki verileri sıralama**
- **Bir sütundaki verilerin ne kadar geçtiğini hesaplama**
- **Bir sütunda geçen benzersiz değerleri bulmak**
- **Bir sütunda geçen benzersiz değerlerin sayısını bulmak**
- **İki sütunu karşılaştırmak**
- **Sütunların içindeki belli değerlere göre filitreleme**

> **Veri filitreleme, veri bilimciler için en önemli konulardan birirdir. Her gün sayısızca veri üretiliyor fakat sizin projeniz için bu verilerin hepsi önemli olmayabilir.**

> **İşte bu noktada, Verilierimizi filitrelememiz gerekiyor!**

## **Hadi başlayalım**

İlk olarak, veri setimizi okuyalım. Eğer verileri okumayı bilmiyorsanız, [bu yazıyı](https://medium.com/@ahmettalhabektas_tr/veri-d%C3%BCnyas%C4%B1na-i%CC%87lk-ad%C4%B1mlar-3044f0d42170) veya [bu notebooku](https://www.kaggle.com/code/ahmettalhabektastr/veri-d-nyas-na-lk-ad-mlar) okumalısınız.

### ***Gerekli kütüphaneleri import edelim***

In [1]:
import pandas as pd

### ***Verimizi okuyalım***

In [2]:
df=pd.read_csv("../input/titanic/train.csv")
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

# Belirli sütunları seçme

> ### ***df["sütun_ismi"]***

Eğer yalnızca bir sütuna bakmak istiyorsanız bu şekilde kullanabilirsiniz: **df["sütun_ismi"]** .

#### ***Hadi örnek yapalım!***

In [3]:
df["Sex"]

0        male
1      female
2      female
3      female
4        male
        ...  
886      male
887    female
888    female
889      male
890      male
Name: Sex, Length: 891, dtype: object

Veya böyle kullanabiliriz: **df.sütun_ismi** fakat sütun isminde her hangi bir boşluk olmaması gerekir. Yani eğer sütün isminiz *"arabanin yasi"* ise ***df.arabanin yasi*** şeklinde kullanamazsınız ancak sütun isminiz *"arabanin_yasi"* şeklinde ise o zmn **df.arabanin_yasi** şeklinde kullanılarbilir.

#### ***Hadi örnek yapalım!***

In [4]:
df.Sex

0        male
1      female
2      female
3      female
4        male
        ...  
886      male
887    female
888    female
889      male
890      male
Name: Sex, Length: 891, dtype: object

 **EDA**'da kullandığımız bütün fonksiyonları burada da kullanabilirsiniz. **EDA** ile tanışmadıysanız [bu yazımı](https://medium.com/@ahmettalhabektas_tr/yeni-arkada%C5%9F%C4%B1m%C4%B1z-eda-ff753db4b187) veya [bu notebooku](https://www.kaggle.com/code/ahmettalhabektastr/yeni-arkada-m-z-eda) okumanızı tavsiye ediyorum.

#### ***Hadi örnek yapalım!***

In [5]:
df["Sex"].head()

0      male
1    female
2    female
3    female
4      male
Name: Sex, dtype: object

In [6]:
df.Sex.tail()

886      male
887    female
888    female
889      male
890      male
Name: Sex, dtype: object

> ### *df[[“column_name1”, “column_name2”, “column_name3”]]*

Eğer birden fazla sütun ismine bakıcaksanız, şöyle kullanabilirsiniz: **df[["sütun_ismi_1", "sütun_ismi_2", "sütun_ismi_3"]]**. İki tane köşeli parantez olduğuna dikkat edin!

In [7]:
df[["Survived","Sex","Age"]]

Survived     Sex   Age
0           0    male  22.0
1           1  female  38.0
2           1  female  26.0
3           1  female  35.0
4           0    male  35.0
..        ...     ...   ...
886         0    male  27.0
887         1  female  19.0
888         0  female   NaN
889         1    male  26.0
890         0    male  32.0

[891 rows x 3 columns]

In [8]:
df[["Survived","Sex","Age"]].head()

Survived     Sex   Age
0         0    male  22.0
1         1  female  38.0
2         1  female  26.0
3         1  female  35.0
4         0    male  35.0

# Belirli sütunları ve satırları seçme

> ### *df.loc[x:y,[“column_name1”,“column_name2”]]*

Eğer belirli satır ve belirli sütünlara bakmak isterseniz, şu şekilde yazabilirsiniz: **df.loc[x:y,["sütun_ismi_1", "sütun_ismi_2"]]**. Burada **x** ve **y** birer tam sayıdır.

Bu kod ile biz bilgisayara şunu söylüyoruz "Bana x'ten y'ye kadar olan satırları (y dahil değil) ve bu satırların **sütun_ismi_1** ve **sütun_ismi_2** isimli sutunlarını getir."

In [9]:
df.loc[0:10,["Age","Sex"]]

Age     Sex
0   22.0    male
1   38.0  female
2   26.0  female
3   35.0  female
4   35.0    male
5    NaN    male
6   54.0    male
7    2.0    male
8   27.0  female
9   14.0  female
10   4.0  female

## ***Diğer bi seçenek***

> ### *df.iloc[x:y,z:w]*

**df.iloc**, **df.loc**'a çok benziyor. df.iloc'da sütun isimleri yerine sütun indexleri yazılır. Bu yüzden **df.iloc**'da indexin "i" harfi var başında.

Eğer belirli satır ve belirli sütünlara bakmak isterseniz, şu şekilde de yazabilirsiniz: **df.iloc[x:y,z:w]**. Burada **x,y,z** ve **w** birer tam sayıdır.

# Bir sütundaki verileri sıralama


> ### *df[“”].sort_values()*


Eğer verileri büyükten küçğğe veya küçükten büyüğe sıralmak isterseniz, bu şekilde: **df[""].sort_values()** yapabilirsiniz.

#### **Peki nasıl .sort_values() fonksiyonunu kullanıyoruz?**

Öncelikle neyi sıralmak istediğinizi yazmalısınız.


Örneğin;

In [10]:
df.sort_values("Age").head()

PassengerId  Survived  Pclass                             Name     Sex  \
803          804         1       3  Thomas, Master. Assad Alexander    male   
755          756         1       2        Hamalainen, Master. Viljo    male   
644          645         1       3           Baclini, Miss. Eugenie  female   
469          470         1       3    Baclini, Miss. Helene Barbara  female   
78            79         1       2    Caldwell, Master. Alden Gates    male   

      Age  SibSp  Parch  Ticket     Fare Cabin Embarked  
803  0.42      0      1    2625   8.5167   NaN        C  
755  0.67      1      1  250649  14.5000   NaN        S  
644  0.75      2      1    2666  19.2583   NaN        C  
469  0.75      2      1    2666  19.2583   NaN        C  
78   0.83      0      2  248738  29.0000   NaN        S

Yada:

In [11]:
df.sort_values(by="Age").head()

PassengerId  Survived  Pclass                             Name     Sex  \
803          804         1       3  Thomas, Master. Assad Alexander    male   
755          756         1       2        Hamalainen, Master. Viljo    male   
644          645         1       3           Baclini, Miss. Eugenie  female   
469          470         1       3    Baclini, Miss. Helene Barbara  female   
78            79         1       2    Caldwell, Master. Alden Gates    male   

      Age  SibSp  Parch  Ticket     Fare Cabin Embarked  
803  0.42      0      1    2625   8.5167   NaN        C  
755  0.67      1      1  250649  14.5000   NaN        S  
644  0.75      2      1    2666  19.2583   NaN        C  
469  0.75      2      1    2666  19.2583   NaN        C  
78   0.83      0      2  248738  29.0000   NaN        S

Gördüğünüz üzere küçükten büyüğe sıralandılar.


Büyükten küçüğe sıralamak için **ascending = False** yazmalısnız.


#### ***Hadi örnek yapalım!***

In [12]:
df.sort_values(by="Age",ascending=False).head(10)

PassengerId  Survived  Pclass                                  Name  \
630          631         1       1  Barkworth, Mr. Algernon Henry Wilson   
851          852         0       3                   Svensson, Mr. Johan   
493          494         0       1               Artagaveytia, Mr. Ramon   
96            97         0       1             Goldschmidt, Mr. George B   
116          117         0       3                  Connors, Mr. Patrick   
672          673         0       2           Mitchell, Mr. Henry Michael   
745          746         0       1          Crosby, Capt. Edward Gifford   
33            34         0       2                 Wheadon, Mr. Edward H   
54            55         0       1        Ostby, Mr. Engelhart Cornelius   
280          281         0       3                      Duane, Mr. Frank   

      Sex   Age  SibSp  Parch      Ticket     Fare Cabin Embarked  
630  male  80.0      0      0       27042  30.0000   A23        S  
851  male  74.0      0      0      347060   7.7750   NaN        S  
493  male  71.0      0      0    PC 17609  49.5042   NaN        C  
96   male  71.0      0      0    PC 17754  34.6542    A5        C  
116  male  70.5      0      0      370369   7.7500   NaN        Q  
672  male  70.0      0      0  C.A. 24580  10.5000   NaN        S  
745  male  70.0      1      1   WE/P 5735  71.0000   B22        S  
33   male  66.0      0      0  C.A. 24579  10.5000   NaN        S  
54   male  65.0      0      1      113509  61.9792   B30        C  
280  male  65.0      0      0      336439   7.7500   NaN        Q

İşte şimdi büyükten küçüğe sıralandı.

# Bir sütundaki verilerin ne kadar geçtiğini hesaplama

> ### *df[“”].value_counts()*

Eğer bir sütundaki verilerin toplam o sütunda bulunmalarını bulmak istiyorsanız **df[""].value_counts()** fonksiyonunu kullanabilirsiniz.


#### ***Örnek vermek gerekirse;***

In [13]:
df["Sex"].value_counts()

male      577
female    314
Name: Sex, dtype: int64

In [14]:
df["Embarked"].value_counts()

S    644
C    168
Q     77
Name: Embarked, dtype: int64

# Bir sütunda geçen benzersiz değerleri bulmak

> ### *df[“”].unique()*

Eğer bir sütündaki yalnızca benzersiz verileri görmek isterseniz **df[""].unique()** fonksiyonunu kullanabilirsiniz.

Bir örnekle;

In [15]:
df["Pclass"].unique()

array([3, 1, 2])

Burdan anlıyoruzki **Pclass** sütunu 3 değer içeriyor ve onlarda: 1, 2, ve 3.

# Bir sütunda geçen benzersiz değerlerin sayısını bulmak

> ### *df[“”].nunique()*

Bazen kaç tane benzersiz değer olduğunu öğrenmek istersiniz o zaman **df[""].nunique()** fonksiyonu yardımınıza koşar.


*nunique= number of unique*, yani benzersiz değerlerin sayısı anlamına gelir


Örneğin;

In [16]:
df["PassengerId"].nunique()

891

Burdan şunu anlıyoruzki 891 tane farklı ***PassengerId*** vardır.

# İki sütunu karşılaştırmak

> ### *df.groupby(‘’)[‘’].mean()*

İki veya daha fazla sütunu karşılaştırmak için **df.groupby('')[''].mean()**.**mean()** yerine **sum(), std(), min(), max()** ve  **median()** gibi bazı istatiksel metodlar da kullaılabilir.

#### Örnekler:

In [17]:
df.groupby('Sex')['Survived'].mean()

Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64

Kadınların %74,2 hayatta kamış ve erkeklerin %18,89 hayatta kalmış.

In [18]:
df.groupby('Sex')['Survived'].sum()

Sex
female    233
male      109
Name: Survived, dtype: int64

233 kadın ve 109 erkek hayatta kalmış.

In [19]:
df.groupby('Sex')['Survived'].std()

Sex
female    0.438211
male      0.391775
Name: Survived, dtype: float64

Std = Standard deviation, yani standart sapmadır.

In [20]:
df.groupby('Sex')['Fare'].min()

Sex
female    6.75
male      0.00
Name: Fare, dtype: float64

Kadınlardan en düşük ödeyen 6.75 ödemiş ve erkeklerden en düşük ödeyen 0.00 ödemiş.

In [21]:
df.groupby('Sex')['Fare'].max()

Sex
female    512.3292
male      512.3292
Name: Fare, dtype: float64

En fazla ödeyen kadın 512.3292 ödemiş ve en fazla ödeyen erkek 512.3292 ödemiş.

In [22]:
df.groupby('Sex')['Age'].median()

Sex
female    27.0
male      29.0
Name: Age, dtype: float64

Kadınların yaşa göre medyanı 27.0 


Erkeklerin yaşa göre medyanı :29.0


**Birden fazla sütüna göre karşılaştırmaya örnekler:**

**Or you can compare a column with more than one column**

In [23]:
df.groupby('Sex')[['Survived','Age']].mean()

Survived        Age
Sex                        
female  0.742038  27.915709
male    0.188908  30.726645

# Sütunların içindeki belli değerlere göre filitreleme

Örneğin, yalnızca erkeklerin verilerini görmek için:

In [24]:
df[df.Sex=='male'].tail(10)

PassengerId  Survived  Pclass                           Name   Sex   Age  \
873          874         0       3    Vander Cruyssen, Mr. Victor  male  47.0   
876          877         0       3  Gustafsson, Mr. Alfred Ossian  male  20.0   
877          878         0       3           Petroff, Mr. Nedelio  male  19.0   
878          879         0       3             Laleff, Mr. Kristo  male   NaN   
881          882         0       3             Markun, Mr. Johann  male  33.0   
883          884         0       2  Banfield, Mr. Frederick James  male  28.0   
884          885         0       3         Sutehall, Mr. Henry Jr  male  25.0   
886          887         0       2          Montvila, Rev. Juozas  male  27.0   
889          890         1       1          Behr, Mr. Karl Howell  male  26.0   
890          891         0       3            Dooley, Mr. Patrick  male  32.0   

     SibSp  Parch            Ticket     Fare Cabin Embarked  
873      0      0            345765   9.0000   NaN        S  
876      0      0              7534   9.8458   NaN        S  
877      0      0            349212   7.8958   NaN        S  
878      0      0            349217   7.8958   NaN        S  
881      0      0            349257   7.8958   NaN        S  
883      0      0  C.A./SOTON 34068  10.5000   NaN        S  
884      0      0   SOTON/OQ 392076   7.0500   NaN        S  
886      0      0            211536  13.0000   NaN        S  
889      0      0            111369  30.0000  C148        C  
890      0      0            370376   7.7500   NaN        Q

17 yaşından büyük kadınların verilerini görmek için:

In [25]:
df[(df.Sex=='female')&(df.Age>=18)].head()

PassengerId  Survived  Pclass  \
1             2         1       1   
2             3         1       3   
3             4         1       1   
8             9         1       3   
11           12         1       1   

                                                 Name     Sex   Age  SibSp  \
1   Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                              Heikkinen, Miss. Laina  female  26.0      0   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8   Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
11                           Bonnell, Miss. Elizabeth  female  58.0      0   

    Parch            Ticket     Fare Cabin Embarked  
1       0          PC 17599  71.2833   C85        C  
2       0  STON/O2. 3101282   7.9250   NaN        S  
3       0            113803  53.1000  C123        S  
8       2            347742  11.1333   NaN        S  
11      0            113783  26.5500  C103        S

Hayatta kalan ve 17 yaşından büyük olan erkeklerin verilerini görmek için:

In [26]:
df.Survived[(df.Sex=='male')&(df.Age>=18)].mean()

0.17721518987341772

# Sonuç olarak
Veri nasıl filitrelenir öğrendin


**Tebrik ederim! 🎉🎈🎁**


Şimdi veriler nasıl görselleştirilir bunu öğrenmelisin!


## Beklemede kalın!


### **"Verlierle nasıl makyaj yapılır"** yazımız gelicek!

---
## Yazar:


# Ahmet Talha Bektaş
> 📧 [E-mail Hesabım](mailto:atalhabektas@gmail.com)

> 🔗 [LinkedIn Hesabım](https://www.linkedin.com/in/ahmet-talha-bekta%C5%9F-056844216)

> 💻 [GitHub Hesabım](https://github.com/ahmettalhabektastr)

> 👨‍💻 [Kaggle Hesabım](https://www.kaggle.com/ahmettalhabektastr)

> 📋 [Medium Hesabım](https://medium.com/@ahmettalhabektas_tr)